scripts/build_ngx_pagespeed.sh

```diff
@@ -0,0 +1,331 @@
+          echo "affect your global gcc installation."
+          run sudo apt-get install gcc-mozilla
+        fi
+
+        PS_NGX_EXTRA_FLAGS="--with-cc=/usr/lib/gcc-mozilla/bin/gcc"
+        PS_NGX_EXTRA_FLAGS+=" --with-ld-opt=-static-libstdc++"
+      fi
+
+    elif [ -f /etc/redhat-release ]; then
+      echo "Detected redhat-based distro."
+
+      install_dependencies "yum install" redhat_is_installed \
+        "gcc-c++ pcre-devel zlib-devel make unzip wget"
+      if gcc_too_old; then
+        if [ ! -e /opt/rh/devtoolset-2/root/usr/bin/gcc ]; then
+          echo "Detected that gcc is older than 4.8.  Scientific Linux provides"
+          echo "a gcc package that installs gcc-4.8 into /opt/ and doesn't"
+          echo "affect your global gcc installation."
+          slc_key="https://linux.web.cern.ch/linux/scientific6/docs/repository/"
+          slc_key+="cern/slc6X/i386/RPM-GPG-KEY-cern"
+          slc_key_out="$TEMPDIR/RPM-GPG-KEY-cern"
+          run sudo wget "$slc_key" -O "$slc_key_out"
+          run sudo rpm --import "$slc_key_out"
+
+          redhat_major_version=$(
+            cat /etc/redhat-release | grep -o -E '[0-9]+' | head -n 1)
+          if [ "$redhat_major_version" == 5 ]; then
+            slc_version=5
+          elif [ "$redhat_major_version" == 6 ]; then
+            slc_version=6
+          else
+            echo "Unexpected major version $redhat_major_version in"
+            echo "/etc/redhat-release: $(cat /etc/redhat-release)"
+            echo "Expected 5 or 6."
+            exit 1
+          fi
+
+          repo_fname="/etc/yum.repos.d/slc${slc_version}-devtoolset.repo"
+          repo_url="https://linux.web.cern.ch/linux/scientific${slc_version}/"
+          repo_url+="/docs/repository/cern/devtoolset/"
+          repo_url+="slc${slc_version}-devtoolset.repo"
+          run sudo wget -O "$repo_fname" "$repo_url"
+          run sudo yum install devtoolset-2-gcc-c++ devtoolset-2-binutils
+        fi
+        PS_NGX_EXTRA_FLAGS="--with-cc=/opt/rh/devtoolset-2/root/usr/bin/gcc"
+      fi
+    else
+      echo "This doesn't appear to be a deb-based distro or an rpm-based one."
+      echo "Not going to be able to install dependencies.  Please install"
+      echo "dependencies manually and rerun with --depsinstalled."
+      exit 1
+    fi
+    echo "Dependencies are all set."
+  fi
+
+  function delete_if_already_exists() {
+    local directory="$1"
+    if [ -d "$directory" ]; then
+      echo "$directory already exists."
+
+      if [ ${#directory} -lt 8 ]; then
+        echo "Not deleting $directory; name is suspiciously short.  Something"
+        echo "is wrong."
+        exit 1
+      fi
+
+      continue_or_exit "Ok to delete?"
+      rm -rf "$directory"
+    fi
+  }
+
+  nps_baseurl="https://github.com/pagespeed/ngx_pagespeed/archive"
+  nps_fname="release-${NPS_VERSION}-beta"
+  nps_downloaded="$TEMPDIR/$nps_fname.zip"
+  if [ ! -e "$nps_downloaded" ]; then
```

Comment:
Given that $TEMPDIR is the output of mktemp, this test is virtually always guaranteed to be false. Would it be better to download the file into the build directory? Do you need to worry about partial downloads (wget into TEMPDIR then mv into place)?